java/ql/lib/semmle/code/java/frameworks/apache/IOGenerated.qll

```diff
@@ -1,684 +1,828 @@
+        "org.apache.commons.io.output;LockableFileWriter;true;LockableFileWriter;(String,boolean);;Argument[0];generated:create-file",
+        "org.apache.commons.io.output;LockableFileWriter;true;LockableFileWriter;(String,boolean,String);;Argument[0];generated:create-file",
+        "org.apache.commons.io.output;XmlStreamWriter;true;XmlStreamWriter;(File);;Argument[0];generated:create-file",
+        "org.apache.commons.io.output;XmlStreamWriter;true;XmlStreamWriter;(File,String);;Argument[0];generated:create-file",
+        "org.apache.commons.io.serialization;ClassNameMatcher;true;matches;(String);;Argument[-1];generated:regex-use[0]",
+        "org.apache.commons.io.serialization;ClassNameMatcher;true;matches;(String);;Argument[0];generated:regex-use[f]",
+        "org.apache.commons.io;CopyUtils;true;copy;(InputStream,Writer);;Argument[0];generated:write-file",
+        "org.apache.commons.io;CopyUtils;true;copy;(InputStream,Writer,String);;Argument[0];generated:write-file",
+        "org.apache.commons.io;CopyUtils;true;copy;(Reader,OutputStream);;Argument[0];generated:write-file",
+        "org.apache.commons.io;CopyUtils;true;copy;(Reader,OutputStream,String);;Argument[0];generated:write-file",
+        "org.apache.commons.io;CopyUtils;true;copy;(Reader,Writer);;Argument[0];generated:write-file",
+        "org.apache.commons.io;CopyUtils;true;copy;(String,OutputStream);;Argument[0];generated:write-file",
+        "org.apache.commons.io;CopyUtils;true;copy;(String,OutputStream,String);;Argument[0];generated:write-file",
+        "org.apache.commons.io;CopyUtils;true;copy;(String,Writer);;Argument[0];generated:write-file",
+        "org.apache.commons.io;CopyUtils;true;copy;(byte[],Writer);;Argument[0];generated:write-file",
+        "org.apache.commons.io;CopyUtils;true;copy;(byte[],Writer,String);;Argument[0];generated:write-file",
+        "org.apache.commons.io;FileSystemUtils;true;freeSpace;(String);;Argument[0];generated:regex-use[f]",
+        "org.apache.commons.io;FileSystemUtils;true;freeSpaceKb;(String);;Argument[0];generated:regex-use[f]",
+        "org.apache.commons.io;FileSystemUtils;true;freeSpaceKb;(String,long);;Argument[0];generated:regex-use[f]",
+        "org.apache.commons.io;FileUtils;true;copyDirectory;(File,File);;Argument[1];generated:create-file",
+        "org.apache.commons.io;FileUtils;true;copyDirectory;(File,File,FileFilter);;Argument[1];generated:create-file",
+        "org.apache.commons.io;FileUtils;true;copyDirectory;(File,File,FileFilter,boolean);;Argument[1];generated:create-file",
+        "org.apache.commons.io;FileUtils;true;copyDirectory;(File,File,FileFilter,boolean,CopyOption[]);;Argument[1];generated:create-file",
+        "org.apache.commons.io;FileUtils;true;copyDirectory;(File,File,boolean);;Argument[1];generated:create-file",
+        "org.apache.commons.io;FileUtils;true;copyDirectoryToDirectory;(File,File);;Argument[1];generated:create-file",
```

Comment:
Above in `copyFileToDirectory` we didn't treat 'copy to directory' (i.e. copy from `arg0` to `arg1/basename(arg0)`) as if they were creating arg1 -- we should treat them alike, probably including them even though `arg1` isn't the exact path created.

Reply:
I don't fully understand, but in any case these models also existed before.